.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,102 @@
+name: Build and push docker images for backend
+
+on:
+  push:
+    branches: [master]
+
+env:
+  DOCKER_REGISTRY: aipipeline
+
+jobs:
+  api-server:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: docker/setup-qemu-action@v1
+      - uses: docker/setup-buildx-action@v1
+      -
+        name: Login to DockerHub
+        uses: docker/login-action@v1 
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      -
+        run: echo "TAG=`echo ${GITHUB_SHA} | cut -c1-7`" >> $GITHUB_ENV
+      -
+        name: Build and Push Image
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          file: backend/Dockerfile
+          tags: ${{ env.DOCKER_REGISTRY }}/api-server:${{ env.TAG }}
```

Comment:
can we have another push to push this image to the `latest` tag? We will only be pushing images for commits that made to the master branch.

Reply:
Thanks @Tomcli. Do we still want both or just latest?

Reply:
you can set tag to latest for now, I think dockerhub added some limit on the number of tag we can have on free tier. We will add the commit tag later on after we figure out a better clean up workflow.